src/viewport.cpp

```diff
@@ -1389,12 +1389,17 @@ static void ViewportAddSignStrings(DrawPixelInfo *dpi, const std::vector<const S
 	if (small) flags.Set(ViewportStringFlag::Small);
 
 	/* Signs placed by a game script don't have a frame. */
-	ViewportStringFlags deity_flags{flags};
+	ViewportStringFlags deity_flags{ flags };
+	deity_flags.Set(ViewportStringFlag::TextColour);
+
 	flags.Set(IsTransparencySet(TO_SIGNS) ? ViewportStringFlag::TransparentRect : ViewportStringFlag::ColourRect);
 
 	for (const Sign *si : signs) {
+		/* Hack to make sure white is actually white. The string drawing turns it into light grey. */
```

Comment:
Is this a temporary solution while you think of another? (Only asking because it says hack, the rest of the code looks fine to me.)

Reply:
I'm a bit in doubt. I could add a parameter to `ViewportDrawStrings` to make it not modify the text color, but that means the sign text colors will be slightly different compared to other strings drawn with the same color. The hack only modifies the behavior for white and keeps everything the same, and the solution is much less intrusive, but certainly isn't a prize winner.

 Maybe I should rebrand it as a workaround ;)